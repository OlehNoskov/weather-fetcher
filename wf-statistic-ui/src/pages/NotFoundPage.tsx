import React from "react";
import {Button} from "@mui/material";
import {useNavigate} from "react-router-dom";

export default function NotFoundPage() {
    const navigate = useNavigate();

    return (
        <div className={"App"}>
            <div className={"not-found"}>
                <h1>Ooops! Page not found!</h1>
                <Button className={"button-home"} variant="contained"
                        onClick={() =>navigate('/')} size={"large"} color="error">
                    Dashboard
                </Button>
            </div>
        </div>
    );
};