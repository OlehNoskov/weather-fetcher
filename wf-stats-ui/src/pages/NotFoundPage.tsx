import React from "react";
import {Button} from "@mui/material";
import {useNavigate} from "react-router-dom";

export default function NotFoundPage() {
    const navigate = useNavigate();
    const navigateDashboard = () => {
        navigate('/');
    };

    return (
        <div className={"App"}>
            <div className={"not-found"}>
                <h1>Ooops! Page not found!</h1>
                <Button className={"button-home"} variant="contained"
                        onClick={navigateDashboard} size={"large"} color="error">
                    Dashboard
                </Button>
            </div>
        </div>
    );
};