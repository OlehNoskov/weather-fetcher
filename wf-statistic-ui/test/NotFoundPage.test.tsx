import {screen, render} from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import NotFoundPage from "../src/pages/NotFoundPage";
import {MemoryRouter} from "react-router-dom";

describe('NotFoundPage', () => {
    it('should render NotFoundPage when type incorrect url', () => {
        const {asFragment} = render(
            <MemoryRouter>
                <NotFoundPage/>
            </MemoryRouter>);

        expect(asFragment()).toMatchSnapshot();
    });

    it('should navigate to DashboardPage after click on \'Dashboard\' button', () => {
        render(
            <MemoryRouter>
                <NotFoundPage/>
            </MemoryRouter>);

        const dashboardButton = screen.getByText('Dashboard');
        userEvent.click(dashboardButton);

        expect(window.location.pathname).toBe('/');
    });
});
