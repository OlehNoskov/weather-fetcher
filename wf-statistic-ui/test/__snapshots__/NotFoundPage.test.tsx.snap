// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage should render DashboardPage after click on Dashboard button 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="not-found"
    >
      <h1
        data-testid="not-found-page"
      >
        Ooops! Page not found!
      </h1>
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge button-home css-1i749hr-MuiButtonBase-root-MuiButton-root"
        href="/"
        tabindex="0"
      >
        Dashboard
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotFoundPage should render NotFoundPage when type incorrect url 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="not-found"
    >
      <h1
        data-testid="not-found-page"
      >
        Ooops! Page not found!
      </h1>
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge button-home css-1i749hr-MuiButtonBase-root-MuiButton-root"
        href="/"
        tabindex="0"
      >
        Dashboard
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotFoundPage should render NotFoundPage when typed incorrect url 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="not-found"
    >
      <h1
        data-testid="not-found-page"
      >
        Ooops! Page not found!
      </h1>
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge button-home css-1i749hr-MuiButtonBase-root-MuiButton-root"
        href="/"
        tabindex="0"
      >
        Dashboard
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotFoundPage should render NotFoundPage when typed incorrect url 2`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="not-found"
    >
      <h1
        data-testid="not-found-page"
      >
        Ooops! Page not found!
      </h1>
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge button-home css-1i749hr-MuiButtonBase-root-MuiButton-root"
        href="/"
        tabindex="0"
      >
        Dashboard
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
